test/base/trainopts.js

```diff
@@ -74,102 +74,78 @@ describe('train() options', () => {
   });
 });
 
+describe('train() and trainAsync() use the same internal methods', () => {
+  
+  it(' _prepTraining()', (done) => {
+    let trainingData = [ { input: [0, 0], output: [0] } ];
+    var net = new brain.NeuralNetwork();
+    let tick = 0;
+    net._prepTraining = (d, o) => {
```

Comment:
Why is this calling itself?  It stack is going to get pretty deep, pretty fast.

Reply:
The tests pass, don't ask!  All we care about are the 
![image](https://user-images.githubusercontent.com/2294139/36323961-0a08c07a-1310-11e8-9c74-7fd41665ce65.png) marks.

Actually I think I am doing it wrong.

Reply:
lol

After talking privately, I believe sinon will be your guid here.  I look forward to seeing this mature.